src/dist/replication/meta_server/greedy_load_balancer.cpp

```diff
@@ -819,6 +833,10 @@ void greedy_load_balancer::greedy_balancer(const bool balance_checker)
 
     for (const auto &kv : apps) {
         const std::shared_ptr<app_state> &app = kv.second;
+        if (in_ignored_apps(kv.first)) {
```

Comment:
I think you can move this code behind of `if (app->status != app_status::AS_AVAILABLE)`